### ISSUE TITLE
Question about training

### ISSUE THREAD

**Issue by Zh-MicroSat** [Chen-Yang-Liu/MLAT#1]
First of all I would like to thank the authors for their code, it is very helpful for my work, I have encountered two problems when trying to reproduce your paper. 
1: I have finished training 17 epochs, but the best-checkpoint is obtained at epoch=2, and the BLUE4 score is only about 0.202 at this time, which is not 0.34 of the BLUE4 score obtained in your article. how do I solve this problem? Can you provide us with your training logs or the Beck-checkpoint model obtained from the training?
2:During my training I noticed that my loss was decreasing at each epoch, but it seems that the final BLUE score did not improve and was always between 0.19 and 0.22 during my training? What is the reason for this situation?
By the way, I would also like to ask how long did it take you to complete the 100 epoch training? I'm using 2080Ti for training and it seems to take 4 days to complete the training, is this reasonable?
Very much looking forward to your answer, thanks again for your perfect work!

**Comment by Chen-Yang-Liu** [comment]
> First of all I would like to thank the authors for their code, it is very helpful for my work, I have encountered two problems when trying to reproduce your paper. 1: I have finished training 17 epochs, but the best-checkpoint is obtained at epoch=2, and the BLUE4 score is only about 0.202 at this time, which is not 0.34 of the BLUE4 score obtained in your article. how do I solve this problem? Can you provide us with your training logs or the Beck-checkpoint model obtained from the training? 2:During my training I noticed that my loss was decreasing at each epoch, but it seems that the final BLUE score did not improve and was always between 0.19 and 0.22 during my training? What is the reason for this situation? By the way, I would also like to ask how long did it take you to complete the 100 epoch training? I'm using 2080Ti for training and it seems to take 4 days to complete the training, is this reasonable? Very much looking forward to your answer, thanks again for your perfect work!

A1: Sorry, due to a long time, the relevant best-trained models and logs are not retained now. According to my impression, the best model will indeed be obtained within 10 epochs. This seems to be a problem with the RSICD data. There is a relatively large difference in the distribution of the training set and the test set of the RSICD dataset.

About your result with 0.22 BLEU, I guess that the beam search strategy is not used. Please try using eval.py to obtain the evaluation metrics.

A2: The evaluation metric and the loss function are not linearly correlated. In my opinion, the maximum epoch set to 30 is enough, more epochs will not bring performance improvement.

**Comment by Zh-MicroSat** [comment]
Thank you for your reply, I plan to retrain again to observe the results. According to your reply, the BLUE4 score of Best-checkpoint shown in the train5.py file is not accurate due to not using beam-search, I really didn't find the beamsearch part in train5.py, I will try to run eval.py to observe the result. Thanks again for your prompt reply.
In addition, I have to point out that there is some problem in the checkpoint loading part of train5.py file, the original code will prompt that "encoder_lr_scheduler" does not exist when loading the checkpoint, I hope you can update it in time to help I hope you can update it to help others. Sorry to bother you late at night, thanks again!

**Comment by rongtongxueya** [comment]
I wonder in what range are these indicators normal in the remote sensing caption data set

**Comment by Chen-Yang-Liu** [comment]
> BLUE4

Thank you for your attention. In my opinion, it is normal that the BLEU4 indicator should be between 0.3 and 0.4 on the RSICD data set.

**Comment by Chen-Yang-Liu** [comment]
> I wonder in what range are these indicators normal in the remote sensing caption data set

Thank you for your attention. In my opinion, it is normal that the BLEU4 indicator should be between 0.3 and 0.4 on the RSICD data set.
